src/sage/categories/finite_groups.py

```diff
@@ -101,12 +101,16 @@ def cardinality(self):
                 sage: G.cardinality()
                 384
             """
+
+            # Note for developers - self will have _deg attribute if it is derived from  MatrixGroup_generic class
+            if hasattr(self, '_deg'):
+                return self._deg
+
             try:
                 o = self.order
+                return o()
```

Comment:
This looks wrong.  The idea of the try except is to check whether `self` has an `order` method.  However, this line of code is from 2010, so it might well be that things could be done better now.